File: Data Structures & Algorithms/balanced-binary-tree/submission-2.cpp
class Solution {
public:
    bool balanced=true;
    int dfs(TreeNode* root){
        if(root==nullptr){
            return 0;
        }
        int leftHeight=dfs(root->left);
        int rightHeight=dfs(root->right);
        if(leftHeight == -1) return -1;
        if(rightHeight == -1) return -1;
        if(abs(leftHeight - rightHeight) > 1){
            return -1;
        }
        return 1 + max(leftHeight, rightHeight);
    }
    bool isBalanced(TreeNode* root) {
        return dfs(root)!=-1;
    }
};

File: Data Structures & Algorithms/two-integer-sum-ii/submission-2.java
class Solution{
    public int[] twoSum(int[] numbers, int target){
        int n=numbers.length,left=0,right=n-1;
        for(int i=0;i<n;i++){
            if(numbers[left]+numbers[right]<target){
                left++;
            }else if(numbers[left]+numbers[right]>target){
                right--;
            }else{
                return new int[]{left+1,right+1};
            }
        }
        return new int[]{};
    }
}